pandaskill/app/leaderboard_page.py

```diff
@@ -0,0 +1,135 @@
+    with setting_columns[1]:
+        date = st.date_input("Date", date_default, key="leaderboard_date")
+    with setting_columns[2]:
+        all_regions_choice = ["All"] + ALL_REGIONS
+        region = st.selectbox("Region", all_regions_choice, 0)
+    with setting_columns[3]:    
+        all_roles_choice = ["All"] + data["role"].unique().tolist()
+        role = st.selectbox("Role", all_roles_choice, 0)
+
+    since = date - dt.timedelta(days=30*6)
+    since = since.strftime("%Y-%m-%d")
+    min_nb_games = 10
+    parameters = {
+        "since": since,
+        "min_nb_games": min_nb_games
+    }
+    data = data.loc[data["date"] <= dt.datetime.combine(date, dt.datetime.min.time())]
+
+    st.info(f"Leaderboard at date {date}, with at least {min_nb_games} games since {since}")
+
+    ranking = create_global_player_ranking(data, parameters)
+
+    average_pscore = data.groupby("player_id")["performance_score"].mean()
+    ranking["pscore"] = ranking["player_id"].map(average_pscore)
+
+
+    if region != "All" or role != "All":
+        if region != "All":
+            ranking = ranking.loc[ranking["region"] == region]
+        if role != "All":
+            ranking = ranking.loc[ranking["role"] == role]
+        ranking["rank"] = range(1, len(ranking) + 1)
+
+    
+    if ranking_type == "Team":
+        ranking = ranking.sort_values("skill_rating", ascending=False) # so that we select the top 5 players per team
+        ranking = ranking.groupby("team_name").agg(
+            region=("region", "first"),
+            nb_games=("nb_games", "mean"),
+            last_game_date=("last_game_date", "max"),
+            pscore=("pscore", "mean"),
+            skill_rating_mu=("skill_rating_mu", lambda x: np.mean(x[:5])),
+            skill_rating_sigma=("skill_rating_sigma", lambda x: np.sqrt(np.mean(np.square(x[:5])))),
+        ).reset_index()
+        ranking["skill_rating"] = ranking["skill_rating_mu"] - 3 * ranking["skill_rating_sigma"]
+        ranking = ranking.sort_values("skill_rating", ascending=False)
+        ranking["rank"] = range(1, len(ranking) + 1)     
+        ranking = ranking.loc[:, ["rank", "team_name", "region", "nb_games", "last_game_date", "pscore", "skill_rating_mu", "skill_rating_sigma", "skill_rating"]]
+    else:   
+        ranking = ranking.loc[:, ["rank", "player_name", "team_name", "role", "region", "nb_games", "last_game_date", "pscore", "skill_rating_mu", "skill_rating_sigma", "skill_rating"]]
+
+    ranking_formatted = ranking.rename(columns={
+        "rank": "Rank",
+        "player_name": "Player",
+        "team_name": "Team",
+        "role": "Role",
+        "region": "Region",
+        "nb_games": "Nb Games",
+        "last_game_date": "Last Game Date",
+        "pscore": "PScore",
+        "skill_rating_mu": "Skill Rating Mu",
+        "skill_rating_sigma": "Skill Rating Sigma",
+        "skill_rating": "Skill Rating (99.7% CI)"
```

Comment:
the 99.7% not explained here could be weird for someone not familiar with stats. I would add a link to https://fr.wikipedia.org/wiki/R%C3%A8gle_68-95-99,7